stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -321,12 +321,14 @@ rmul!(A::AbstractMatrix, D::Diagonal) = @inline mul!(A, A, D)
 lmul!(D::Diagonal, B::AbstractVecOrMat) = @inline mul!(B, D, B)
 
 function (*)(A::AdjOrTransAbsMat, D::Diagonal)
```

Comment:
The implementation is identical to the generic ones in lines 311ff and 315ff. Do we need these methods at all (like for disambiguation)?

Reply:
Awesome! Making the world a better place by removing code. 🤣